### ISSUE TITLE
CMake: install target missing

### ISSUE THREAD

**Issue by davydden** [cp2k/dbcsr#45]
I am sure you are aware of this, but CMake build lacks install target. 

Probably, this should be (in case of a general out-of-source build):

- [ ] `<CMAKE_BUILD_DIR>/src/*.mod -> prefix/include/*`
- [ ] `<CMAKE_BUILD_DIR>/src/*.a|so|dylib -> prefix/lib/*`
- [ ] `<CMAKE_SOURCE_DIR>/src/*.h -> prefix/include/*`

**Comment by dev-zero** [comment]
yes, I am aware. Unfortunately I didn't have the time yet to write that one (PR appreciated ;-))

**Comment by davydden** [comment]
that's usually done via
```
SET(TARGETLIBEXPORT "dbcsr")
# 1. Install defined library target
install (TARGETS ${TARGETLIB} EXPORT ${TARGETLIBEXPORT}
         LIBRARY DESTINATION lib)
# 2. Install content of include dir
install (DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/include/"
         DESTINATION include)
```
however I never did anything with Fortran in Cmake. So I keep this code snippet here in case it helps later.

**Comment by dev-zero** [comment]
thanks, fixed it with #48
